### PR TITLE
[1591] added belongs_to association for lead_school for trainee

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

https://trello.com/c/OZG89cfc/1591-s-create-association-between-trainees-and-schools

Should merge after PR https://github.com/DFE-Digital/register-trainee-teachers/pull/785

### Changes proposed in this pull request

* new column "lead_school_id" added to trainees
* belongs_to association added to Trainee for School (optional)

### CODE REVIEW

**Comment by mooktakim** [comment]
https://trello.com/c/OZG89cfc/1591-s-create-association-between-trainees-and-schools

**Comment by mooktakim** [comment]
[[S] Create association between trainees and schools](https://trello.com/c/OZG89cfc/1591-s-create-association-between-trainees-and-schools)

File: app/models/trainee.rb [hunk 1]
```diff
@@ -10,6 +10,7 @@ class Trainee < ApplicationRecord
   has_many :nationalities, through: :nationalisations
   has_many :trainee_disabilities, dependent: :destroy, inverse_of: :trainee
   has_many :disabilities, through: :trainee_disabilities
+  belongs_to :lead_school, optional: true, class_name: "School"
```
Comment (darokel) [inline app/models/trainee.rb:13]: I'm still not sure whether this is the best design. Doing it like this means that this field (and `employing_school` if we follow the pattern) will be null for potentially thousands of trainee records. Did you try using a join table and compare the two approaches? (we already do that for disabilities, nationalities, subjects etc)

Comment (darokel) [reply app/models/trainee.rb:13]: Discussed and voted offline. Consensus is to go with foreign key

**Review by darokel — APPROVED** [review]
Looks good but just needs that other PR in so we can uncomment the assertion in the spec

**Pull request merged**